File: ts/wam/5074.ts
export const WamWebDbTableUsage = 5074 as const

export interface WebDbTableUsageEvent {
  /** field 1, wire `offline_session_id` */
  offlineSessionId?: string
  /** field 2, wire `web_scenario` */
  webScenario?: WEBSCENARIOCODE
  /** field 3, wire `web_table` */
  webTable?: string
  /** field 4, wire `web_table_log_reason` */
  webTableLogReason?: WEBTABLELOGREASONCODE
  /** field 5, wire `web_table_read_count` */
  webTableReadCount?: number
  /** field 6, wire `web_table_write_count` */
  webTableWriteCount?: number
}

File: rust/wam/5074.rs
pub const WAM_WEB_DB_TABLE_USAGE: u32 = 5074;

#[derive(Debug, Default)]
pub struct WebDbTableUsageEvent {
    /// field 1, wire `offline_session_id`
    pub offline_session_id: Option<String>,
    /// field 2, wire `web_scenario`
    pub web_scenario: Option<WEBSCENARIOCODE>,
    /// field 3, wire `web_table`
    pub web_table: Option<String>,
    /// field 4, wire `web_table_log_reason`
    pub web_table_log_reason: Option<WEBTABLELOGREASONCODE>,
    /// field 5, wire `web_table_read_count`
    pub web_table_read_count: Option<i64>,
    /// field 6, wire `web_table_write_count`
    pub web_table_write_count: Option<i64>,
}